include/mbgl/shaders/mtl/collision.hpp

```diff
@@ -0,0 +1,202 @@
+#pragma once
+
+#include <mbgl/shaders/collision_layer_ubo.hpp>
+#include <mbgl/shaders/shader_source.hpp>
+#include <mbgl/shaders/mtl/shader_program.hpp>
+
+namespace mbgl {
+namespace shaders {
+
+#define COLLISION_SHADER_COMMON \
```

Comment:
Could you get rid of the macro? Maybe use a constexpr `std::string` instead?

Reply:
I tried something but it's not working to merge strings in initialization like this.

Reply:
I mean here: `static constexpr auto source = COLLISION_SHADER_COMMON R"(`

Reply:
Hmm seems like std::string can actually be constexpr, but only short strings.

Trying to figure out how we can concat constexpr `std::string_view`s

Reply:
@alexcristici https://github.com/alexcristici/maplibre-native/pull/29

Reply:
Looks like a good solution, but I would say let's merge this PR as it is, and make this change in the follow up PR.

Reply:
@alexcristici OK sounds good to me.